File: math1.py
import math

def isOddEven(n):
    if n % 2 == 0:
        return 0
    else:
        return 1
    
def isPrime(n):
    if n <= 1:
        return 0
    elif n == 2:
        return 1
    elif n % 2 == 0:
        return 1
    
    for i in range(3, int(math.sqrt(n)), 2):
        if n % (i * i) == 0:
            return 0
        
    return 1
    
def minSumSeq(n):   # kx = n - (k * (k - 1)) / 2  # n: sum, k: count, x: first term
    for k in range(2, n):
        kx = n - (k * (k - 1)) // 2
        if kx % k == 0:
            x = kx // k
            if x > 0:
                result = [str(x + i) for i in range(k)]
                return', '.join(result)

n = int(input('number\n'))
print(isPrime(n))

File: math1.c
#include <stdio.h>
#include <stdlib.h>

int isOddEven(int n) {
    if (n % 2 == 0)
        return 0;
    else
        return 1;
}

int isPrime(int n) {
    if (n <= 1)
        return 0;
    else if (n == 2)
        return 1;
    else if (n % 2 == 0)
        return 0;
    
    for (int i = 3; i * i < n; i++) {
        if (n % (i * i) == 0)
            return 0;
    }
    
    return 1;
}

char* minSumSeq(int n) {    // kx = n - (k * (k - 1)) / 2  # n: sum, k: count, x: first term
    for (int k = 2; k < n; k++) {
        int kx = n - (k * (k - 1)) / 2;
        if (kx % k == 0) {
            int x = kx / k;
            if (x > 0) {
                char *result = malloc(n * 2);
                int pos = 0;
                for (int i = 0; i < k; i++) {
                    pos += sprintf(result + pos, "%d", x + i);
                    if (i < k - 1) {
                        pos += sprintf(result + pos, ", ");
                    }
                }
                
                return result;
            }
        }
    }

    return NULL;
}

void main() {
}
